client/src/components/TaskCard/TaskCard.js

```diff
@@ -21,6 +21,10 @@ const TaskCard = ({
     setModal(!showModal);
   };
 
+  const toggleReflection = () => {
+    setReflection(!showReflection);
```

Comment:
I would also rename the state to be `setShowReflectionModal` and `showReflectionModal`

Reply:
Renamed.